### PR TITLE
feat(list): supplement some list cmds, and coresponding tests

### PR DESCRIPTION
**Pull request by MciG-ggg**
## Summary by CodeRabbit

* **New Features**
  * Added four list operations: LPUSHX/RPUSHX, LINSERT (BEFORE/AFTER), and RPOPLPUSH; exposed a Before/After selector.

* **Bug Fixes**
  * Improved per-key locking and concurrency safety, consistent empty-result behavior, safe pop limits, and cross-key validation for pop-and-push.

* **Tests**
  * Expanded coverage for new list commands, concurrency, atomicity, and many edge cases.

* **Chores**
  * Updated repository ignore patterns.

### CODE REVIEW

**Review by AlexStocks — APPROVED** [review]

**Comment by guozhihao-224** [comment]
It appears that no key locking operation was performed, which could lead to data corruption in the metadata.

**Review by MciG-ggg — COMMENTED** [review]
> It appears that no key locking operation was performed, which could lead to data corruption in the metadata. 

So should i add rwlock or sth else to the concurrent critical section?

File: src/storage/src/storage_impl.rs [hunk 1]
```diff
@@ -422,6 +440,31 @@ impl Storage {
...
+        value: &[u8],
+    ) -> Result<i64> {
+        let slot_id = key_to_slot_id(key);
+        let instance_id = self.slot_indexer.get_instance_id(slot_id);
+        self.insts[instance_id].linsert(key, before_or_after, pivot, value)
+    }
+    pub fn rpoplpush(&self, source_key: &[u8], destination_key: &[u8]) -> Result<Option<Vec<u8>>> {
+        let source_slot_id = key_to_slot_id(source_key);
+        let dest_slot_id = key_to_slot_id(destination_key);
+        if source_slot_id != dest_slot_id {
```
Comment (guozhihao-224) [inline src/storage/src/storage_impl.rs]: - Cross-slot errors should not be exposed to the client; the implementation in CPP can be referenced： https://github.com/arana-db/kiwi-cpp/blob/0fa4f1b95ff2f89873d06eb86de40532dd91d146/src/storage/src/storage.cc#L946 
- However, I believe the CPP implementation also has problems. The semantics of the RPOPLUSH command are atomic; the involved keys should be locked when the command is issued to ensure atomicity.

Comment (MciG-ggg) [reply src/storage/src/storage_impl.rs]: 1. Ok, i get it, i will fix it
2. I have read the implementation of CPP version, i see that it split RPoplpush to RPop and LPush, and it does not lock any of these two commands. But in my implementation, in my opinion, i lock both commands, you can check it at https://github.com/MciG-ggg/kiwi/blob/main/src/storage/src/redis_lists.rs#L1021
